src/cargo/core/compiler/build_context/target_info.rs

```diff
@@ -208,6 +210,14 @@ impl TargetInfo {
             process.arg("--print=crate-name"); // `___` as a delimiter.
             process.arg("--print=cfg");
 
+            if gctx.cli_unstable().check_target_cfgs {
+                process.arg("--print=crate-name"); // `___` as a delimiter.
+                process.arg("--print=check-cfg");
+
+                process.arg("--check-cfg=cfg()"); // otherwise `--print=check-cfg` won't output
```

Comment:
Why is that?

Reply:
The comment is a bit misleading, `--print=check-cfg` alone would print something, just not what we want, without `--check-cfg=cfg()` it would be like we didn't enable check-cfg checking and so `--print=check-cfg` would only return the default state (of no checking).

We also have `--check-cfg=cfg()` when generating the [`check_cfg_args`](https://github.com/rust-lang/cargo/blob/a9e8aa2c65091b520519fd1065f41d156d45f9dd/src/cargo/core/compiler/mod.rs#L1500).